scripts/process_uploads.py

```diff
@@ -3,89 +3,63 @@
 Processes a new batch of samples from the UPLOAD bucket,
 into the MAIN and ARCHIVE buckets using the batch_move_files function. 
 
-Determines the files to be moved through a comparison of the 
-most recently updated CSV and the CSV uploaded in a previous batch run. 
+Assumes that the csv file has been processed in a previous step. 
 """
 
 import os
-import io
-import csv
-from os.path import basename, join
-from typing import List, Optional, Set, Tuple
+from os.path import join
+from typing import List, Optional, Tuple
 import click
 import hailtop.batch as hb
-from google.cloud import storage
-from joint_calling.upload_processor import batch_move_files, SampleGroup
-
 
-def samples_from_csv(bucket_name, prefix) -> Tuple[List[str], str]:
-    """Determines list of samples from a csv file within a specified bucket
+# Import SampleAPI
+from sample_metadata.api.sample_api import SampleApi
+from joint_calling.upload_processor import batch_move_files, SampleGroup
 
-    Assumptions
-    ==========
-    Only one .csv file exists in the upload bucket and previous batch directory.
-    """
 
-    client = storage.Client()
-    bucket = client.get_bucket(bucket_name)
+# Reading CSV pulled out into separate function.
 
-    all_blobs = list(client.list_blobs(bucket_name, prefix=prefix))
-    csv_path = next(filter(lambda blob: blob.name.endswith('.csv'), all_blobs)).name
 
-    blob = bucket.get_blob(csv_path).download_as_text()
-    csv_reader = csv.DictReader(io.StringIO(blob))
+def get_samples_from_db() -> List[str]:
+    """ Pulls list of samples to be moved from SampleMetadata DB """
 
+    sapi = SampleApi()
     samples: List[str] = []
-    for row in csv_reader:
-        samples.append(row['sample.sample_name'])
-
-    return samples, csv_path
 
+    # Returns a list of all samples with status == uploaded
+    # Alternatively, could return all samples that were not included in the latest .mt analysis.
+    samples = sapi.get_latest_uploads()  # TO IMPLEMENT: API CALL
```

Comment:
We want to be a bit clearer here, a sample doesn't have a status (because it's a physical status). The status = uploaded would be a byproduct of either the gvcf or cram being uploaded, so this call might look like:

- `get_samples_which_have_gvcfs(project)`